vertical-pod-autoscaler/pkg/recommender/model/cluster.go

```diff
@@ -292,15 +302,17 @@ func (cluster *ClusterState) aggregateStateKeyForContainerID(containerID Contain
 // findOrCreateAggregateContainerState returns (possibly newly created) AggregateContainerState
 // that should be used to aggregate usage samples from container with a given ID.
 // The pod with the corresponding PodID must already be present in the ClusterState.
-func (cluster *ClusterState) findOrCreateAggregateContainerState(containerID ContainerID) *AggregateContainerState {
+func (cluster *ClusterState) findOrCreateAggregateContainerState(containerID ContainerID, isPodActive bool) *AggregateContainerState {
 	aggregateStateKey := cluster.aggregateStateKeyForContainerID(containerID)
 	aggregateContainerState, aggregateStateExists := cluster.aggregateStateMap[aggregateStateKey]
 	if !aggregateStateExists {
 		aggregateContainerState = NewAggregateContainerState()
 		cluster.aggregateStateMap[aggregateStateKey] = aggregateContainerState
 		// Link the new aggregation to the existing VPAs.
 		for _, vpa := range cluster.Vpas {
-			vpa.UseAggregationIfMatching(aggregateStateKey, aggregateContainerState)
+			if vpa.UseAggregationIfMatching(aggregateStateKey, aggregateContainerState) && isPodActive {
+				cluster.VpasWithMatchingPods[vpa.ID] = true
```

Comment:
Why are we checking isPodActive here?
This way we'll mark [Cron]Job-related VPA as not matching pods quite often, won't we?

Reply:
Yes, this is something I'm not entirely sure about. For short running pods this doesn't work too well. On the other hand, for deployments, if we count terminated pods, after deployment scales down to 0 it will take us 8 days to start saying that there are no matching pods. I'm open to discussion on this.

Reply:
Agreed to not take into account if pod is active. Check removed.